private/bufpkg/bufcheck/bufcheckserver/internal/buflintvalidate/field.go

```diff
@@ -700,6 +755,284 @@ func checkTimestampRules(adder *adder, timestampRules *validate.TimestampRules)
 	return nil
 }
 
+func checkExampleValues(
+	adder *adder,
+	pathToExampleValues []int32,
+	fieldConstraints *validate.FieldConstraints,
+	typeRulesMessage protoreflect.Message,
+	containingMessageDescriptor protoreflect.MessageDescriptor,
+	// Not nil only if fieldDescriptor is a synthetic field for map key/value.
+	parentMapFieldDescriptor protoreflect.FieldDescriptor,
+	fieldDescriptor protoreflect.FieldDescriptor,
+	exampleValues []protoreflect.Value,
+	extensionTypeResolver protoencoding.Resolver,
+) error {
+	if err := protoencoding.ReparseExtensions(extensionTypeResolver, typeRulesMessage); err != nil {
```

Comment:
This feels like a pretty deep level to be calling protoencoding.ReparseExtensions, to do this per field seems unnecessary, can you provide some documentation as to why we're calling this here, and why it can't be called at a higher level?

Reply:
Add a comment to clarify this, but basically it's to ensure that predefined rules from imported files are properly resolved.

Reply:
This comment doesn't make sense to me as a reader - I am still confused why I need to call `ReparseExtensions` for every field. Can you make sure this comment would make sense to another reader without context by verifying with someone else what their understanding of it is?